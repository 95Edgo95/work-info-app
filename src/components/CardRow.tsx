import Icon from "components/Icon";
import * as React from "react";

import * as styles from "assets/css/cardRow.css";

interface IWorkbookInfoRowProps {
  value: string | number;
  tooltip: string;
  icon: string;
}

export default function CardRow({ icon, value, tooltip }: IWorkbookInfoRowProps) {
  return (
    <div className={styles.detail}>
      <Icon name={icon} title={tooltip} />
      <span className={styles.detailValue}>{value}</span>
    </div>
  );
}
