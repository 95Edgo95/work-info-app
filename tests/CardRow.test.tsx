import CardRow from "../src/components/CardRow";
import { mount } from "enzyme";
import * as React from "react";

describe('CardRow', () => {
  it('should render correctly', () => {
    const component = mount(
      <CardRow value="test" icon="trash" />
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should render correctly with number', () => {
    const component = mount(
      <CardRow value={2} icon="trash" />
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
