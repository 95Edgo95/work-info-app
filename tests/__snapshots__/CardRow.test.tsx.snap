// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRow should render correctly 1`] = `
<CardRow
  icon="trash"
  value="test"
>
  <div>
    <Icon
      name="trash"
    >
      <span
        className="undefined "
        role="presentation"
        style={
          Object {
            "color": "black",
            "fontSize": "24px",
          }
        }
      >
        <svg>
          <use
            xlinkHref="#trash"
          />
        </svg>
      </span>
    </Icon>
    <span>
      test
    </span>
  </div>
</CardRow>
`;

exports[`CardRow should render correctly with number 1`] = `
<CardRow
  icon="trash"
  value={2}
>
  <div>
    <Icon
      name="trash"
    >
      <span
        className="undefined "
        role="presentation"
        style={
          Object {
            "color": "black",
            "fontSize": "24px",
          }
        }
      >
        <svg>
          <use
            xlinkHref="#trash"
          />
        </svg>
      </span>
    </Icon>
    <span>
      2
    </span>
  </div>
</CardRow>
`;
